backend/src/test/java/org/tim/services/ReportServiceTests.java

```diff
@@ -0,0 +1,189 @@
+package org.tim.services;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.lang.StringUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.tim.entities.Message;
+import org.tim.entities.Project;
+import org.tim.entities.Translation;
+import org.tim.exceptions.EntityNotFoundException;
+import org.tim.repositories.MessageRepository;
+import org.tim.repositories.ProjectRepository;
+import org.tim.repositories.TranslationRepository;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Locale;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.tim.constants.CSVFileConstants.CSV_FILE_NAME;
+
+@ExtendWith(MockitoExtension.class)
+public class ReportServiceTests {
+
+    @Mock
+    ProjectRepository projectRepository;
+    @Mock
+    MessageRepository messageRepository;
+    @Mock
+    TranslationRepository translationRepository;
+    @Mock
+    Message message1;
+    @Mock
+    Message message2;
+
+    private Locale testLocale1 = new Locale("PL");
+    private Locale testLocale2 = new Locale("US");
+    private Locale testLocale3 = new Locale("UK");
+    private Locale testLocale4 = new Locale("RUS");
+
+    @Test
+    void generateExcelReport_projectNotFound_exception() {
+        //given
+        Mockito.lenient().when(projectRepository.findById(1L)).thenReturn(Optional.empty());
+        var reportService = new ReportService(projectRepository, messageRepository, translationRepository);
+        //when, then
+        assertThrows(EntityNotFoundException.class, () -> {
+            reportService.generateCSVReport(1L, new String[]{});
+        });
+    }
+
+
+    @Test
+    void generateExcelReport_validCall_correctCSVFileProduced() {
+        //given
+        testInitialize();
+        var reportService = new ReportService(projectRepository, messageRepository, translationRepository);
+        //when
+        reportService.generateCSVReport(0L, new String[]{testLocale1.toString(), testLocale2.toString(), testLocale3.toString(), testLocale4.toString()});
+        //then
+        try {
+            var reader = Files.newBufferedReader(Paths.get(CSV_FILE_NAME));
+            var parser = new CSVParser(reader, CSVFormat.DEFAULT);
+            var records = parser.getRecords();
+            //Locales
+            assertEquals("Locale", records.get(0).get(0));
+            assertEquals("pl", records.get(1).get(0));
+            assertEquals("pl", records.get(2).get(0));
```

Comment:
In all tests we decided to use assertAll like:
assertAll(
		() -> assertFalse(actual2.isPresent()),
		() -> assertFalse(actual3.isPresent()),
);
This solution first run all tests and then will inform about failed asserations. Withouth this first broken assert will broke whole test.